src/ucp/core/ucp_am.c

```diff
@@ -998,18 +980,17 @@ UCS_PROFILE_FUNC(ucs_status_ptr_t, ucp_am_send_nbx,
     /* Note that max_eager_short.memtype_on is always initialized to real
      * max_short value
      */
-    max_short = ucp_ep_config(ep)->am_u.max_eager_short.memtype_on;
-
     if (flags & UCP_AM_SEND_REPLY) {
-        ret = ucp_am_send_req(req, count, &ucp_ep_config(ep)->am, param,
-                              ucp_ep_config(ep)->am_u.reply_proto,
-                              ucs_min(max_short, UCP_AM_SHORT_REPLY_MAX_SIZE),
-                              flags);
+        max_short = ucp_ep_config(ep)->am_u.max_reply_eager_short.memtype_on;
+        proto     = ucp_ep_config(ep)->am_u.reply_proto;
     } else {
-        ret = ucp_am_send_req(req, count, &ucp_ep_config(ep)->am, param,
-                              ucp_ep_config(ep)->am_u.proto, max_short, flags);
+        max_short = ucp_ep_config(ep)->am_u.max_eager_short.memtype_on;
+        proto     = ucp_ep_config(ep)->am_u.proto;
```

Comment:
duplication of what ucp_am_try_send_short is doing

Reply:
Do you mean getting short threshold?
here I also need to init `proto`. I'd have to have a branch for it anyway

Reply:
can we get the max_short/proto before ucp_am_try_send_short , and pass it to ucp_am_try_send_short as input param?